src/status_im/contexts/wallet/wallet_connect/core.cljs

```diff
@@ -85,12 +85,19 @@
 
 (defn required-networks-supported?
   [proposal supported-networks]
-  (let [required-networks (get-in proposal [:params :requiredNamespaces :eip155 :chains])
-        supported-eip155  (set (map chain-id->eip155 supported-networks))]
-    (every? #(contains? supported-eip155 %) required-networks)))
+  (let [required-namespaces (get-in proposal [:params :requiredNamespaces])]
+    (when (= (keys required-namespaces) '(:eip155))
+      (let [required-networks (get-in required-namespaces [:eip155 :chains])
+            supported-eip155  (set (map chain-id->eip155 supported-networks))]
+        (every? #(contains? supported-eip155 %) required-networks)))))
 
 (defn get-networks-by-mode
   [db]
   (let [test-mode? (get-in db [:profile/profile :test-networks-enabled?])
         networks   (get-in db [:wallet :networks (if test-mode? :test :prod)])]
     (mapv #(-> % :chain-id) networks)))
+
+(defn event-should-be-handled?
```

Comment:
This function also fixes an issue raised here: https://github.com/status-im/status-mobile/pull/20693#issuecomment-2228288723

TL;DR; If a user connected an app on `testnet` and then switched to `mainnet`. A WC request sent on `testnet` will show up and fail. 

But it needs one more clause, if the event is not meant to be handled, it should be rejected. So WC can send new events in future. 

This is the apt error response: 
```  USER_REJECTED_CHAINS: {
    message: "User rejected chains.",
    code: 5001,
  },
  ```
  
  In this function you can do something like: 
  
  ```clj
(defn event-should-be-handled? []
  (let [to-handle? (some ...)]
    (when-not to-handle?
      (rf/dispatch [:respond-with-rejection] ()))
    to-handle?)) 
```

Here's an example on how to send a response: 
```clj
(rf/reg-event-fx
 :wallet-connect/reject-session-request
 (fn [_ _]
   {:fx [[:dispatch
          [:wallet-connect/send-response
           {:error (wallet-connect/get-sdk-error
                    constants/wallet-connect-user-rejected-error-key)}]]]}))
```

The constant in this case will be `USER_REJECTED_CHAINS`

Reply:
it's probably worth creating a separate PR, otherwise this one will get postponed for some time again :) so feel free to raise it